src/main.rs

```diff
@@ -37,57 +38,53 @@ mod pci;
 mod pe;
 mod virtio;
 
-#[cfg(not(test))]
-#[panic_handler]
-#[allow(clippy::empty_loop)]
-fn panic(_info: &PanicInfo) -> ! {
-    loop {}
+global_asm!(include_str!("asm/ram64.s"));
+
+extern "C" {
+    fn halt_loop() -> !;
 }
 
-#[cfg(not(test))]
-/// Enable SSE2 for XMM registers (needed for EFI calling)
-fn enable_sse2() {
-    unsafe {
-        asm!("movq %cr0, %rax");
-        asm!("or $$0x2, %ax");
-        asm!("movq %rax, %cr0");
-        asm!("movq %cr4, %rax");
-        asm!("or $$0x600, %ax");
-        asm!("movq %rax, %cr4");
-    }
+#[cfg_attr(not(test), panic_handler)]
+fn panic(info: &PanicInfo) -> ! {
+    log!("PANIC: {}", info);
+    unsafe { halt_loop() }
 }
 
-#[cfg(not(test))]
 /// Setup page tables to provide an identity mapping over the full 4GiB range
 fn setup_pagetables() {
     const ADDRESS_SPACE_GIB: u64 = 64;
```

Comment:
FYI, we don't need this to be 64 for Cloud Hypervisor any more (4GiB is sufficient)

Reply:
Awesome! I'll change it to 4, do you know why it needed to be 4GiB?